libcontainer/rootfs_linux_test.go

```diff
@@ -35,3 +39,79 @@ func TestCheckMountRoot(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestNeedsSetupDev(t *testing.T) {
+	config := &configs.Config{
+		Mounts: []*configs.Mount{
+			{
+				Device:      "bind",
+				Source:      "/dev",
+				Destination: "/dev",
+			},
+		},
+	}
+	setupDev, err := needsSetupDev(config)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if setupDev {
+		t.Fatalf("expected false")
+	}
+}
+
+func TestNeedsSetupDevStrangeSource(t *testing.T) {
+	config := &configs.Config{
+		Mounts: []*configs.Mount{
+			{
+				Device:      "bind",
+				Source:      "/devx",
+				Destination: "/dev",
+			},
+		},
+	}
+	setupDev, err := needsSetupDev(config)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if setupDev {
+		t.Fatalf("expected false")
```

Comment:
Not familiar with the runc test suite, but It would be helpful to provide a full error message on these fatal calls like `expected setupDev to be false was true`